extensions/amp-document-recommendations/0.1/amp-document-recommendations.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-document-recommendations-0.1.css';
+import {Layout} from '../../../src/layout';
+import {MultidocManager} from '../../../src/runtime';
+import {Services} from '../../../src/services';
+import {assertConfig} from './config';
+import {isJsonScriptTag} from '../../../src/dom';
+import {setStyle} from '../../../src/style';
+import {tryParseJson} from '../../../src/json';
+import {user} from '../../../src/log';
+
+/** @const */
+const MAX_ARTICLES = 2;
+
+/**
+ * @const
+ * TODO(emarchiori): Make this a configurable parameter.
+ */
+const SEPARATPOR_RECOS = 3;
+
+/** @private {AmpDocumentRecommendations} */
+let activeInstance_ = null;
+
+export class AmpDocumentRecommendations extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    // TODO(emarchiori): Consider using a service instead of singleton.
+    if (activeInstance_) {
+      return;
+    }
+    activeInstance_ = this;
+
+    /** @private {?AmpDocumentRecommendationsConfig} */
+    this.config_;
+
+    /** @private {MultidocManager} */
+    this.multidocManager_;
+
+    /** @private {number} */
+    this.nextArticle_ = 0;
+
+    /** @private @const {!../../../src/service/viewer-impl.Viewer} */
+    this.viewer_ = Services.viewerForDoc(this.getAmpDoc());
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER;
```

Comment:
nit: ===

Reply:
Keeping == for consistency with other extensions and rest of the code.

Reply:
we had a long discussions about this in the team :) although style guides says `===`, we are sticking with `==`